src/wp-includes/option.php

```diff
@@ -364,6 +364,157 @@ function get_options( $options ) {
+	 */
+	$old_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s", $option ) );
+	if ( ! $old_autoload || $old_autoload === $autoload ) {
+		return false;
+	}
+
+	// Update the autoload value.
+	$result = $wpdb->update(
+		$wpdb->options,
+		array( 'autoload' => $autoload ),
+		array( 'option_name' => $option )
+	);
+	if ( ! $result ) {
+		return false;
+	}
+
+	// Update caches as necessary based on the autoload change.
+	$alloptions = wp_load_alloptions( true );
+	if ( 'no' === $autoload ) {
+		// If set in alloptions, remove and migrate to individual cache.
+		if ( isset( $alloptions[ $option ] ) ) {
+			wp_cache_set( $option, $alloptions[ $option ], 'options' );
+			unset( $alloptions[ $option ] );
+			wp_cache_set( 'alloptions', $alloptions, 'options' );
+		}
+	} else {
+		// If set in cache, migrate to alloptions and delete individual cache.
+		$value = wp_cache_get( $option, 'options' );
+		if ( false !== $value ) {
+			$alloptions[ $option ] = $value;
+			wp_cache_delete( $option, 'options' );
+			wp_cache_set( 'alloptions', $alloptions, 'options' );
+		}
+	}
+
+	return true;
+}
+
+/**
+ * Sets the autoload value for multiple options in the database.
+ *
+ * Autoloading too many options can lead to performance problems, especially if the options are not frequently used.
+ * This function allows modifying the autoload value for multiple options without changing the actual option value.
+ * This is for example recommended for plugin activation and deactivation hooks, to ensure any options exclusively used
+ * by the plugin which are generally autoloaded can be set to not autoload when the plugin is inactive.
+ *
+ * @since 6.4.0
+ * @see wp_set_option_autoload()
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array       $options  List of option names. Expected to not be SQL-escaped.
+ * @param string|bool $autoload Autoload value to control whether to load the options when WordPress starts up.
+ *                              Accepts 'yes'|true to enable or 'no'|false to disable.
+ * @return array Associative array of all provided $options as keys and boolean values for whether their autoload value
+ *               was updated.
+ */
+function wp_set_options_autoload( $options, $autoload ) {
+	global $wpdb;
+
+	$autoload = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+
+	$results = array();
+	foreach ( $options as $option ) {
+		$results[ $option ] = false;
+	}
+
+	/*
+	 * Determine the relevant options that do not already use the given autoload value.
+	 * If no options are returned, no need to update.
+	 */
+	$placeholders      = trim( str_repeat( '%s,', count( $options ) ), ',' );
+	$options_to_update = $wpdb->get_col( $wpdb->prepare( "SELECT option_name FROM $wpdb->options WHERE autoload != %s AND option_name IN ($placeholders)", $autoload, ...$options ) );
+	if ( ! $options_to_update ) {
+		return $results;
+	}
```

Comment:
Related to my comment in the `wp_set_option_autoload` function, I'm unsure what the value is in this check since we're not avoiding any queries by filtering out the list of passed options before attempting the update. Can you help me understand what you're trying to avoid here and why it's needed?

Reply:
@joemcgill Good point, from a querying perspective this does not avoid any queries. However, it looks like without this query we won't be able to determine which options' autoload values were updated, since the `UPDATE ...` query call does not provide us with that information (it only returns the numbers of rows that were updated, but not which ones). While in your comment above that would not be an issue since it's only for a single option, here it matters.

Given that this is a DB writing function, it wouldn't be called regularly, and therefore not be an area where we would have to pay _as_ close attention to DB queries made. I think it's okay to keep the extra DB call due to the above benefit, as without it the return value would be less helpful.